ce/api/percentileanomaly.py

```diff
@@ -183,62 +291,66 @@ def canonical_timestamp(timestamp, timescale, timeidx):
-                                   timescale, timestamp))
-                else:
-                    baseline = 0.0
-                values = np.asarray([float(v) for v in projected_data[timescale][timestamp].values()])
-                if values.size < 12:
-                    abort(500,
-                          "Not all models available for {} {}. Models available: {}").format(
-                               timescale, timestamp, 
-                               projected_data[timescale][timestamp].keys())
-                elif values.size > 12:
-                    abort(500,
-                          "Extraneous data for {} {}. Models available: {}").format(
-                               timescale, timestamp,
-                               projected_data[timescale][timestamp].keys())
-                anomalies = values - baseline
-                projected_data[timescale][timestamp] = list(
-                    np.percentile(anomalies, percentiles))
+        for p in projected_data:
+            # determine the baseline, if applicable
+            if calculate_anomaly:
+                baseline = None
+                date_format = '%Y-%m-%d %H:%M:%S'
+                projected_date = datetime.strptime(p["date"], date_format)
+
+                for b in baseline_data:
+                    baseline_date = datetime.strptime(b["date"], date_format)
+                    if p["timescale"] == b["timescale"] and baseline_date.month == projected_date.month:
+                        if not baseline:
+                            b["values"] =b["values"][0]
+                            baseline = float(b["values"])
+                        else:
+                            abort(500,
+                                    ("Multiple matching baseline datasets ",
+                                    "for {} {} {} {}").format(
+                                        baseline_model,
+                                        baseline_climatology,
+                                        p["timescale"], p["date"]))       
+
+                if(not baseline):
+                    abort(500, "Missing baseline data: {} {} {}".format(
+                                                baseline_model, baseline_climatology, p["timescale"]))    
+
+            else:
+                baseline = 0.0
+            values = np.asarray([float(v) for v in p["values"]])
+
+            if values.size < 12:
+                abort(500,
+                        "Not all models available for {} {}. Models available: {}").format(
+                            p["timescale"], p["date"], p["models"])
+            elif values.size > 12:
+                abort(500,
+                        "Extraneous data for {} {}. Models available: {}").format(
+                            p["timescale"], p["date"], p["models"])
+
+            anomalies = values - baseline
+
+            p["values"] = list(np.percentile(anomalies, percentiles))
+            del(p["models"])
```

Comment:
We do this because we don't want the "models" key in the final result?

Reply:
Yes, it is. I did it because the original `percentilanomaly` outputs the nested dictionaries without having the `models` attribute in the very inner loop. Should I keep it to have the attribute?